auth/services/oauth/interface.go

```diff
@@ -55,3 +63,15 @@ type Verifier interface {
 	// ClientMetadataURL constructs the URL to the client metadata of the local verifier.
 	ClientMetadataURL(webdid did.DID) (*url.URL, error)
 }
+
+// Holder implements the OpenID4VP Holder role which acts as Authorization server in the OpenID4VP flow.
+type Holder interface {
+	// BuildPresentation builds a Verifiable Presentation based on the given presentation definition.
+	BuildPresentation(ctx context.Context, walletDID did.DID, presentationDefinition pe.PresentationDefinition, verifierMetadata oauth.AuthorizationServerMetadata, nonce string) (*vc.VerifiablePresentation, *pe.PresentationSubmission, error)
+	// ClientMetadata returns the metadata of the remote verifier.
+	ClientMetadata(ctx context.Context, endpoint string) (*oauth.AuthorizationServerMetadata, error)
+	// PostError posts an error to the verifier. If it fails, an error is returned.
```

Comment:
ClientMetadata should return `iam.OAuthClientMetadata`?
This also affects BuildPresentation. Maybe just pass clientMeta.VPFormats?